front/src/index.html

```diff
@@ -141,40 +131,27 @@
                         // TODO: do whatever
                         console.debug(`create room\n title: ${roomData.title}\n desc: ${roomData.description}`);
                         // temporary code to make it look like how it should behave but that really isn't how it'll be done
-                        this.rooms.push({
-                            _id: {
-                                $oid: new Date().getTime().toString(16)
-                            },
-                            title: roomData.title,
-                            description: roomData.description
-                        });
+                        srv.createRoom(roomData.title, roomData.description,[this.loggedUser.id]);
                         this.roomEditorHidden = true;
+                        this.fetchRooms();
                     },
                     joinRoom(id) {
                         // TODO: do whatever
                         console.debug(`join room '${id}'`);
                         // temporary code to make it look like how it should behave but that really isn't how it'll be done
-                        this.rooms.push({
-                            _id: {
-                                $oid: id
-                            },
-                            title: "title",
-                            description: ""
-                        });
+                        addUserToRoom(this.loggedUser.id, id);
+                        this.fetchRooms();
                         this.roomEditorHidden = true;
                     },
                     onLogin(pseudo) {
-                        // TODO: do whatever
-                        console.debug(`logged in as user '${pseudo}'`);
-                        // temporary code to make it look like how it should behave but that really isn't how it'll be done
-                        this.loggedUser = {
-                            "_id": {
-                                "$oid": new Date().getTime().toString(16)
-                            },
-                            "pseudo": pseudo,
-                            "roomsId": []
-                        };
-                        this.users.push(this.loggedUser);
+                        const self = this;
+                        srv.getUserByPseudo(pseudo).then(v => {
+                            this.loggedUser = v
+                        }).catch(e => {
+                            srv.createUser(pseudo, "").then(v => {
+                                self.loggedUser = v
+                            });
+                        });
```

Comment:
maybe add a `UserExists(UserID) => boolean` to simplify this (and make it arround 10x time more readable)